test/unit/tasks/publishing_api_test.rb

```diff
@@ -0,0 +1,384 @@
+
+        test "republishes all documents with HMTL attachments" do
+          edition = create(:published_publication, :with_html_attachment)
+
+          PublishingApiDocumentRepublishingWorker.expects(:perform_async_in_queue).with(
+            "bulk_republishing",
+            edition.document_id,
+            true,
+          )
+          task.invoke
+        end
+      end
+
+      describe "#document_type" do
+        let(:task) { Rake::Task["publishing_api:bulk_republish:document_type"] }
+
+        test "republishes all documents from document type" do
+          edition = create(:publication)
+
+          PublishingApiDocumentRepublishingWorker.expects(:perform_async_in_queue).with(
+            "bulk_republishing",
+            edition.document_id,
+            true,
+          )
+          task.invoke("Publication")
+        end
+      end
+
+      describe "#by_organisation" do
+        let(:org) { create(:organisation) }
+        let(:task) { Rake::Task["publishing_api:bulk_republish:by_organisation"] }
+
+        test "Republishes the latest edition for each document owned by the organisation" do
+          edition = create(:published_news_article, organisations: [org])
+
+          PublishingApiDocumentRepublishingWorker.expects(:perform_async_in_queue).with(
+            "bulk_republishing",
+            edition.document.id,
+            true,
+          )
+
+          task.invoke(org.slug)
+        end
+
+        test "Ignores documents owned by other organisation" do
+          some_other_org = create(:organisation)
+          edition = create(:published_news_article, organisations: [some_other_org])
+
+          PublishingApiDocumentRepublishingWorker.expects(:perform_async_in_queue).with(
+            "bulk_republishing",
+            edition.document.id,
+            true,
+          ).never
+
+          task.invoke(org.slug)
+        end
+      end
+
+      describe "#documents_by_content_ids" do
+        let(:task) { Rake::Task["publishing_api:bulk_republish:documents_by_content_ids"] }
+
+        test "Republishes documents by content ids" do
+          edition = create(:publication)
+
+          PublishingApiDocumentRepublishingWorker.expects(:perform_async_in_queue).with(
+            "bulk_republishing",
+            edition.document.id,
+            true,
+          )
+
+          task.invoke([edition.content_id])
+        end
+      end
+
+      describe "#documents_by_content_ids_from_csv" do
+        let(:task) { Rake::Task["publishing_api:bulk_republish:documents_by_content_ids_from_csv"] }
+
+        test "Republishes documents by content ids from csv" do
+          edition = create(:publication)
+          filename = "content_ids_#{Time.zone.now.to_i}"
+
+          File.open("lib/tasks/#{filename}.csv", "w+")
+          CSV.open("lib/tasks/#{filename}.csv", "wb") do |csv|
+            csv << %w[content_id]
+            csv << [edition.content_id]
+          end
```

Comment:
Might be nicer to use a string for this, something like:

```ruby
File.write("lib/tasks/#{filename}.csv", "content_id\n#{edition.content_id}")
```

Reply:
Nice suggestion, done!